lib/client.js

```diff
@@ -20,58 +20,82 @@ module.exports = function (dependencies) {
   function Client (options) {
     this.config = config(options);
     this.healthCheckInterval = setInterval(() => {
-         if (this.session && !this.session.destroyed) {
-             this.session.ping((error, duration) => {
-                 if (error) {
-                     logger("No Ping response after " + duration + " ms");
-                 }
-                 logger("Ping response after " + duration + " ms");
-             })
-         }
-     }, this.config.heartBeat).unref();
+      if (this.session && !this.session.closed && !this.session.destroyed && !this.isDestroyed) {
+        this.session.ping((error, duration) => {
+          if (error) {
+            logger("No Ping response after " + duration + " ms");
+          }
+          logger("Ping response after " + duration + " ms");
+        });
+      }
+    }, this.config.heartBeat).unref();
   }
 
   Client.prototype.write = function write (notification, device, count) {
+    if (this.isDestroyed) {
```

Comment:
Probably a useful addition to avoid edge cases

Reply:
Yes. I think it is.